src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -1452,6 +1492,34 @@ private void PlaceIfPossible(string organelleType, int q, int r, int rotation, r
         }
     }
 
+    private List<Hex> GetIslandHexes()
+    {
+        var organelles = editedMicrobeOrganelles.Organelles;
+        var initHex = organelles[0].Position;
+        var checkedHexes = new List<Hex> { initHex };
+        var shouldBeChecked = organelles.Select(p => p.Position).ToList();
+
+        CheckmarkNeighbors(checkedHexes, initHex);
+        return shouldBeChecked.Except(checkedHexes).ToList();
+    }
+
+    private void CheckmarkNeighbors(List<Hex> @checked, Hex me)
+    {
+        var myNeighbors = GetNeighborHexes(me).Where(p => !@checked.Contains(p)).ToArray();
+        @checked.AddRange(myNeighbors);
```

Comment:
I'd like to see a couple of comments / xml documentation on what these methods are as these have very dense logic.